src/main/java/io/cdap/plugin/http/HTTPArgumentSetter.java

```diff
@@ -73,8 +76,10 @@ public void run(ActionContext context) throws Exception {
 
         LOG.debug("Request to {} resulted in response code {}.", conf.getUrl(), responseCode);
         if (responseCode != 200) {
-          throw new IllegalStateException(String.format("Received non-200 response code %d. Response message = %s",
-                                                        responseCode, response.getResponseMessage()));
+          String errorReason = String.format("Received non-200 response code %d. Response message = %s",
+                  responseCode, response.getResponseMessage());
+          throw ErrorUtils.getProgramFailureException(new ErrorCategory(ErrorCategory.ErrorCategoryEnum.PLUGIN),
+                  errorReason, errorReason, ErrorType.USER, true, null);
```

Comment:
use `ErrorUtils#getActionErrorByStatusCode`, also add `errorCode` & `errorCodeType` in the `ProgramFailureException`.